lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -713,15 +713,17 @@ FIRRTLModuleLowering::lowerModule(FModuleOp oldModule, Block *topLevelModule,
   if (failed(lowerPorts(firrtlPorts, ports, oldModule, loweringState)))
     return {};
 
-  loweringState.processRemainingAnnotations(oldModule,
-                                            AnnotationSet(oldModule));
   // Build the new hw.module op.
   auto builder = OpBuilder::atBlockEnd(topLevelModule);
   auto nameAttr = builder.getStringAttr(oldModule.getName());
   auto newModule =
       builder.create<hw::HWModuleOp>(oldModule.getLoc(), nameAttr, ports);
   if (auto outputFile = oldModule->getAttr("output_file"))
     newModule->setAttr("output_file", outputFile);
+  if (AnnotationSet::removeAnnotations(oldModule, dutAnnoClass))
+    newModule->setAttr("DesignUnderTest", builder.getUnitAttr());
```

Comment:
This is a firrtl specific annotation being added to a hw module.  Please name the attribute something like `firrtl. DesignUnderTest` (instead of plain DesignUnderTest).  Also, please add a new "FIRRTL specific attributes applied to HW Modules" top-level section to docs/FIRRTLAnnotations.md  and describe this here.

Reply:
🤔 I was originally thinking the same thing: that this logic should just preserve the annotation through lower to HW because it's FIRRTL specific.  However, the notion of "this is the DUT" or "this is the testbench" are probably more general hardware concepts than that and may make sense promoting to a non-FIRRTL prefixed attribute.

What do you think?

Reply:
Here's what I offer to you: if it is prefixed with `firrtl.` then you can get away with a specification of "this is defined as Chisel's notion of what this thing is".  If you make it a general attribute on module, then I will ask for a formalism of what the attribute means, what it affects, etc because it is now a first-class IR extension. :-)

Reply:
😂 😂 😂 Prefix it is.

Reply:
Prefixed the attribute with `firrtl` and updated the doc.

Reply:
This is going to be fairly fragile.  It is really saying "this has public visibility, don't muck with the interfaces" and "this has some tag for easy lookup"  the attribute only addresses the latter.